source/javascripts/components/PipelinesPage/providers/SearchParamsProvider.tsx

```diff
@@ -0,0 +1,40 @@
+import { Dispatch, PropsWithChildren, SetStateAction, createContext, useEffect, useMemo, useState } from 'react';
+
+type Props = PropsWithChildren;
+
+type Context = {
+  searchParams: URLSearchParams;
+  setSearchParams: Dispatch<SetStateAction<URLSearchParams>>;
+};
+
+export const SearchParamsContext = createContext<Context>({
+  searchParams: new URLSearchParams(),
+  setSearchParams: () => undefined,
+});
+
+const SearchParamsProvider = ({ children }: Props) => {
+  const [searchParams, setSearchParams] = useState(new URLSearchParams(window.location.hash.split('?')[1] || ''));
+
+  const value = useMemo(() => {
+    return { searchParams, setSearchParams };
+  }, [searchParams]);
+
+  useEffect(() => {
+    const prevSearchParams = new URLSearchParams(window.location.hash.split('?')[1] || '');
+    const nextSearchParams = new URLSearchParams(searchParams);
+
+    prevSearchParams.sort();
+    nextSearchParams.sort();
+
+    if (prevSearchParams.toString() !== nextSearchParams.toString()) {
```

Comment:
Why do not use lodash's `isEqual` here instead?

Reply:
I don't think it would work because isEqual returns false when two objects contain methods with different references.